src/helpers/rectangle.ts

```diff
@@ -0,0 +1,28 @@
+import { Rect, Zone } from "../types";
+import { intersection } from "./zones";
+
+/**
+ * Compute the intersection of two rectangles. Returns nothing if the two rectangles don't overlap
+ */
+export function rectIntersection(rect1: Rect, rect2: Rect): Rect | undefined {
+  return zoneToRect(intersection(rectToZone(rect1), rectToZone(rect2)));
+}
+
+function rectToZone(rect: Rect): Zone {
```

Comment:
Not technically a zone since we still are in pixels and not in col/row coordinates. But this allow me to use all the helpers of Zone.

Reply:
:man_shrugging: